contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -205,7 +211,8 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         _resetContext();
 
         // Reset the ovmStateManager.
-        ovmStateManager = iOVM_StateManager(address(0));
+        // @todo: should we move this into reset context?
+        ovmStateManager = iOVM_StateManager(MAX_ADDRESS);
```

Comment:
Let's discuss because now that I look more closely, that would also have the effect of resetting the SM up here:

https://github.com/ethereum-optimism/contracts/blob/2a29fe559f3a11ce8d7f5c9b6c7e8990c0569f89/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L185-L191

I'm also just a bit unclear about the purpose of resetting the context there, even prior to this change.